### ISSUE TITLE
Inconsistent close time after hovering toast

### ISSUE THREAD

**Issue by VandalPaulius** [ealush/butter-toast#32]
I have experienced inconsistent close time after hovering mouse over the toast: sometimes my the timeout progress animation almost gets to finish, sometimes toast closes in the half of the animation. (Animation stops onMouseEnter and continues onMouseLeave).

**Comment by ealush** [comment]
Hey @VandalPaulius, thanks for reaching out. 👍 

If I understand you correctly, what you are experiencing is not a bug, but expected (even if annoying) behavior.

Let me explain:
When building developing Butter Toast, I decided that hovering the toast will only prevent it from closing, but not pause its countdown. The mismatch you are experiencing is between your own countdown calculation which pauses, and the toast's inner countdown which doesn't.

It seems reasonable enough that Butter-Toast would behave the way you expect, and it can be added as an option in a future release.

In the meantime, to handle it, you could raise your toast on sticky mode, and whenever **your** countdown ends, call the toast's dismiss function.

It should look somewhat like this:

```js
ButterToast.raise({
// ...
content: ({{toastId, dismiss}}) => <YourAwesomeToast dismiss={dismiss}/>
// ...
});

class YourAwesomeToast extends React.Component {
// ...
    onCountDownEnd() {
        this.props.dismiss();
    }
// ...
}
```

Please let me know if that works for you.

**Comment by ealush** [comment]
Hey @VandalPaulius, I just published 1.1.0 which adds your request as part of the API.
More specifically: `pauseOnHover`. If set to true, for as long as you are hovering the toast, it will not count down.

Use it like this:
```js
<ButterToast pauseOnHover trayPosition="bottom-left"/>
```
Or alternatively:
```js
<ButterToast pauseOnHover={true} trayPosition="bottom-left"/>
```

**Comment by VandalPaulius** [comment]
Works well. Thanks! :)
